lib/inc/drogon/CacheMap.h

```diff
@@ -202,6 +207,8 @@ class CacheMap
                 size_t timeout = 0,
                 std::function<void()> timeoutCallback = std::function<void()>())
     {
+        fnOnInsert_(key);
```

Comment:
Check if it's empty before calling it. 
```c++
if(fnOnInsert_)
    fnOnInsert_(key);
```

Reply:
By default it is initialized with an empty function to avoid checking if is null or not. Anyway no issues at all if you may want to check. Also if you have a better way to bring those events at application level it is welcome. Right now I'm using the branch session_events since my application need to know when a session have been created or deleted. I guess you can modify also my branch. Let me know if I can help.

Reply:
My bad, update the default with the following 
std::function<void(const T1&)> fnOnInsert = [](const T1&){}

Reply:
I did some quick benchmarking. Checking if function is empty then call is faster then calling empty function. (on the order of ns). I prefer check and call, but the performance benefit is very tiny.

![圖片](https://user-images.githubusercontent.com/8792393/197199833-d2f1e21b-0de2-4969-8fe4-8b839e04df13.png)

Reply:
Updated both SessionManager and CacheMap accordingly with your test results. Also moved SessionEvents to an abstract implementation due to the fact we do not need anymore a default empty implementation.

Reply:
btw I also tried to update style following your modifications.